pkg/osbuild/librepo_source.go

```diff
@@ -0,0 +1,120 @@
+package osbuild
+
+import (
+	"fmt"
+
+	"github.com/osbuild/images/pkg/rpmmd"
+)
+
+// LibrepoSource wraps the org.osbuild.librepo osbuild source
+type LibrepoSource struct {
+	Items   map[string]*librepoSourceItem `json:"items"`
+	Options *librepoSourceOptions         `json:"options"`
+}
+
+func NewLibrepoSource() *LibrepoSource {
+	return &LibrepoSource{
+		Items: make(map[string]*librepoSourceItem),
+		Options: &librepoSourceOptions{
+			Mirrors: make(map[string]*librepoSourceMirror),
+		},
+	}
+}
+
+// AddPackage adds the given *depsolved* pkg to the downloading. It
+// needs the *depsovled* repoConfig so that the repoID of the two can
+// be matched up
+func (source *LibrepoSource) AddPackage(pkg rpmmd.PackageSpec, repos []rpmmd.RepoConfig) error {
+	pkgRepo := findRepoById(repos, pkg.RepoID)
+	if pkgRepo == nil {
+		return fmt.Errorf("cannot find repo-id %v for %v in %+v", pkg.RepoID, pkg.Name, repos)
```

Comment:
The `repos` array can contain a lot of text because of gpg keys and long URLs.  Maybe we should make this error message easier to read by printing only the identifiable information from the repos array?  Maybe just the ID and mirror URLs.

Perhaps not worth the effort though.

Reply:
Done